### DISCUSSION TITLE
Agregar nuevo atributo a los cmb-box para agregar mas campos de la tabla en el cmb-box

### DISCUSSION THREAD

**Discussion by rafaelposadaf** [unix4you2/practico#31] [Ideas]
Actualmente en los cmb-box permite solo agregar value y el label, pero a veces se requiere saber otros campos de ese mismo registro cuando el usuario selecciona un elemento del cmb-box; para poder obtener mas información de ese registro se debe realizar una consulta adicional en algunos casos sobre esa misma tabla 

Seria muy interesante poder seleccionar que campos requiero adicionales con el fin de que cuando seleccionen un elemento de la lista poder acceder a dichos valores y no realizar una consulta adicional

Esto se podria solucionar agregando un campo vacio que permita digitar que otros campos adicionales requiere y que estos campos queden en un atributo del select ejemplo: <option data-json="{campo1: valor, campo2: valor2}" value="value1">texto1</option>
Luego en el momento que se cree cada opción se le agrega en ese atributo un json con dichos campos, incluso podria ser con todos los campos de la consulta para mayor practicidad

**Comment by unix4you2** [comment]
Alguna vez me hablaste de algo similar sobre atributos data-xxxxxxx  me regalas info general?  porque pienso que el campo no sólo lo podríamos dejar en los select, sino también en todo campo que venga desde BD.   Cómo son los casos de uso básicos del data-xxxxx ?

**Comment by rafaelposadaf** [comment]
hola John, lo llegue a utilizar en algun desarrollo por la forma como trabajabamos, al pareceer html permite agregar atributos adicionales idependientes de los reservados, es como agregar un atributo adicional solo que por convención se usa "data-id" o "data-xxx" u otro nombre que le quieras dar
https://www.youtube.com/watch?v=vXuSV1S31aU

**Comment by unix4you2** [comment]
OK. Tengo el conflicto revisando el tema...
De dónde saldría el valor?  porque agregar un campo para dar nombres personalizados a los atributos como data-xxxx es simple; pero el tema es si será un valor fijo o si se traerá desde BD, porque en el último caso es como si un control tuviese doble vínculo!
O lo ves más como algo de valor fijo y manipulable posteriormente con JS?

**Reply by rafaelposadaf** [reply]
Lo 2, la idea es que como ya tienes la consulta cuando estas creando el control le agregues un atributo adicional como data-xxxx, o PCO_data etc, y ahi pongas el resultado de la consulta de ese combo box para a travez de js se acceda a es información por cada registro.
Desde la creación del control del combobox seria un campo adicional para agregar mas columnas a la consulta, y el resultado de esa consulta quede en ese data-xxx, para acceder luego a esa información

**Reply by unix4you2** [reply]
Comprendido.  Pero entonces creo que sí podemos dejarlo como un campo extra que exprese "una lista" de campos y que luego se convierta dinámicamente de esa manera sirve no sólo para las listas, sino también para los demás tipos de campo.  Ej:
Nuevo campo LISTA_CAMPOS_DATA = idioma,color,pais

Sobre el control quedarían varios como  data-idioma="xxxx" data-color="yyyy" data-pais="zzzz"

De esa manera creo se puede generalizar a quien necesite un campo o varios de una vez.

Un abrazo
